modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -107,7 +107,8 @@ def _read(cls, filepath_or_buffer, **kwargs):
         # Max number of partitions available
         num_partitions = NPartitions.get()
         # This is the number of splits for the columns
-        num_splits = min(len(column_names), num_partitions)
+        cols_number = len(column_names)
+        num_splits = min(cols_number if cols_number else 1, num_partitions)
```

Comment:
```suggestion
        num_splits = min(len(column_names) or 1, num_partitions)
```

Reply:
Why is the minimum number of columns - 1?

Reply:
This is done to avoid division by zero [here](https://github.com/modin-project/modin/blob/master/modin/data_management/utils.py#L37) (for full traceback see issue header).